runtime/server/timeseries_api.go

```diff
@@ -442,6 +442,8 @@ func (s *Server) GenerateTimeSeries(ctx context.Context, request *api.GenerateTi
 
 func convertRowsToTimeSeriesValues(rows *drivers.Result, rowLength int) ([]*api.TimeSeriesValue, error) {
 	results := make([]*api.TimeSeriesValue, 0)
+	defer rows.Close()
```

Comment:
Instead of calling this in a nested function, it's best to call it after the call that opened the rows (query)

Reply:
changed

Reply:
For this case, you don't want to use `defer` – instead, you directly call `rows.Close()` after you consumed it (just make sure you also call `Close` in the error cases).

`defer` is just a helper that calls a statement when the scope ends

Reply:
Yeah, for such a case, you have to call `Close()` in all the relevant exit cases – or run the query in a narrower scope